src/mono-runtimes/mono-runtimes.targets

```diff
@@ -280,26 +282,41 @@
     </ItemGroup>
   </Target>
 
-  <Target Name="_BuildRuntimes"
-      Condition=" '@(_MonoRuntime)' != '' Or '@(_MonoCrossRuntime)' != '' "
-      DependsOnTargets="_GetRuntimesOutputItems"
-      Outputs="@(_RuntimeSource);@(_RuntimeBinarySource);@(_CrossRuntimeBinarySource);@(_ProfilerSource);@(_MonoPosixHelperSource);@(_RuntimeEglibHeaderSource);@(_MonoBtlsSource);@(_BclTestOutput)">
+  <Target Name="_DownloadArchive"
+      DependsOnTargets="_GetMonoGitCommitHash"
+      Condition=" !Exists('$(MonoSourceFullPath)\sdks\out\.stamp-$(_MonoArchiveName)-download') " >
+    <MakeDir
+        Directories="$(MonoSourceFullPath)\sdks\out"
+    />
+    <DownloadUri
+        SourceUris="https://xamjenkinsartifact.blob.core.windows.net/mono-sdks/$(_MonoArchiveName).tar.gz"
+        DestinationFiles="$(AndroidToolchainCacheDirectory)\$(_MonoArchiveName).tar.gz"
+        ContinueOnError="True"
+    />
     <Exec
-        Command="make DISABLE_IOS=1 $(MakeConcurrency) @(_MonoRuntime->'package-android-%(Identity)', ' ') @(_MonoCrossRuntime->'package-android-%(Identity)', ' ') $(_MonoSdksParameters)"
+        Condition=" Exists('$(AndroidToolchainCacheDirectory)\$(_MonoArchiveName).tar.gz') "
+        Command="tar -xvf $(AndroidToolchainCacheDirectory)\$(_MonoArchiveName).tar.gz"
         IgnoreStandardErrorWarningFormat="True"
-        WorkingDirectory="$(MonoSourceFullPath)\sdks\builds"
+        WorkingDirectory="$(MonoSourceFullPath)\sdks\out"
+    />
+    <Touch
+        Condition=" Exists('$(AndroidToolchainCacheDirectory)\$(_MonoArchiveName).tar.gz') "
+        Files="$(MonoSourceFullPath)\sdks\out\.stamp-$(_MonoArchiveName)-download"
+        AlwaysCreate="True"
     />
+  </Target>
+
+  <Target Name="_Build"
+      Condition=" '@(_MonoRuntime)' != '' Or '@(_MonoCrossRuntime)' != '' Or '@(_MonoBcl)' != '' Or '@(_LlvmRuntime)' != '' "
```

Comment:
This `Condition` is a tad odd: why is `'@(_MonoBcl)' != ''` in here?  The `@(_MonoBcl)` declaration is not conditional, i.e. it's *always* defined, so unless you imagine updating things in the future so that `@(_MonoBcl)` is *empty*...why does `@(_MonoBcl)` need to be checked?

Reply:
Just to match what's already there. If you want I can fix that in a follow-up PR, but I would prefer not to have to wait for a full re-run just for that.